pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py

```diff
@@ -0,0 +1,91 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Optional, Tuple
+
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class FxValidator:
+
+    def check_logging(self, fx_name: str, on_step: bool, on_epoch: bool) -> None:
+        allowed = self.functions[fx_name]
```

Comment:
what happens if we add a hook and forget adding it to this dictionary? this wiill enforce that calling self.log from within that hook will throw, which might not be clear to users

Reply:
If a function is forgotten, I think it should error out and not silently return as we want to enforce that this is properly updated.

If you prefer, this could raise a `RuntimeError` asking to open an issue as it should never happen.

Reply:
I think raising an explicit error message in this case would be good